src/lib.rs

```diff
@@ -32,9 +32,16 @@ pub async fn run_trin(
     info!("Launching Trin: v{trin_version}");
     info!(config = %trin_config, "With:");
 
+    // Setup temp trin data directory if we're in ephemeral mode
+    if trin_config.ephemeral {
+        setup_temp_dir()?;
+    }
+    // Configure node data dir based on the provided private key
+    let private_key = configure_node_data_dir(trin_config.private_key)?;
```

Comment:
It's very easy to miss the side-effect here: that `setup_temp_dir` alters the trin data path environment variable, and that it needs to happen before the `configure_node_data_dir` call, which reads that same variable.

An ideal solution would probable not use the environment variable as like a sneaky global value that every function shares. Instead, we might do a little more logic here that generates the base directory once, and passes it around after that, maybe in the PortalnetConfig. Specifically, this should solve the problem where we can't run multiple `--ephemeral` nodes, concurrently.

But if that doesn't look straightforward to you, then at the very least: let's add a warning comment at the usage here, and also in both the setup and configure methods.